demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -137,17 +131,23 @@ public void onReceived(List<DownloadBatchStatus> downloadBatchStatuses) {
         bindViews();
     }
 
-    VersionOneDatabaseCloner.CloneCallback cloneCallback = new VersionOneDatabaseCloner.CloneCallback() {
+    private final VersionOneDatabaseCloner.CloneCallback cloneCallback = new VersionOneDatabaseCloner.CloneCallback() {
         @Override
         public void onUpdate(String updateMessage) {
             databaseCloningUpdates.setText(updateMessage);
         }
     };
 
-    Migrator.Callback migrationCallback = new Migrator.Callback() {
+    private final DownloadMigrationService.MigrationCallback migrationCallback = new DownloadMigrationService.MigrationCallback() {
         @Override
-        public void onUpdate(MigrationStatus migrationStatus) {
-            databaseMigrationUpdates.setText(migrationStatus.status().toRawValue());
+        public void onUpdate(final MigrationStatus migrationStatus) {
```

Comment:
`onUpdate` is marked with `@WorkerThread` to highlight to clients that they need to handle the thread swap should they want to display anything when an update is received.

Reply:
❌ the updates should be received in the thread where the tool is being used, same as in the downloadmanager service callbacks

Reply:
I agree with @zegnus - the clients of the library shouldn't need to care about managing threads like this.